Training/Program.cs

```diff
@@ -3,31 +3,202 @@
-      #endregion
+using System.Text;
+using static System.Console;
+namespace Training;
 
-      #region Method ------------------------------------------------
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+#region Class Program --------------------------------------------------------------------------
+internal class Program {
+   #region Method ------------------------------------------------
+   static void Main (string[] args) {
+      Wordle w = new ();
+      for (; ; )
+         w.Run ();
+   }
+   #endregion
+}
+#endregion
+
+#region Class Wordle ---------------------------------------------------------------------------
+public class Wordle {
+   #region Constructors ------------------------------------------
+   /// <summary>Static constructor to read and close dict files.</summary>
+   static Wordle () {
+      mPuzzleWords = File.ReadAllLines ("C:/etc/Puzzle-5.txt");
+      mDict = File.ReadAllLines ("C:/etc/Dict-5.txt");
+      CursorVisible = false;
+   }
+   static string[] mPuzzleWords, mDict;
+
+   /// <summary>Create new Wordle Console Game</summary>
+   public Wordle () => mInput = new ConsoleKey[5];
+   ConsoleKey[] mInput;
+   #endregion
+
+   #region Method ------------------------------------------------
+   /// <summary>Run current instance of game.</summary>
+   public void Run () {
+      OutputEncoding = new UnicodeEncoding ();
+      mCursor = 0; mGuessNum = 6; mCorrectGuess = null;
+      mSecretWord = mPuzzleWords[mRandomWord.Next (mPuzzleWords.Length)].ToCharArray ();
+      Clear ();
+      WriteLine ("WORDLE");
+      DisplaySkeleton ();
+      do {
+         GetInput ();
+         mGuessNum--;
+      } while (mGuessNum > 0 && mCorrectGuess == false);
+      NextLine ();
+      ResetColor ();
+      Message (mCorrectGuess == true ?
+               $"Congrats! You got it right in just {6 - mGuessNum} guess{(mGuessNum == 5 ? "" : "es")}." :
+               $"Oops! The answer was {string.Join ("", mSecretWord)}. Better luck next time!");
+      mState.Clear ();
+   }
+   static Random mRandomWord = new ();
+   char[] mSecretWord;
+   bool? mCorrectGuess;
+   int mGuessNum;
+   #endregion
+
+   #region Implementation ----------------------------------------
+   /// <summary>Checks if input is a valid word from dictionary,
+   /// compares input with generated Secret Word, and re-prints coloured input as hints.</summary>
+   /// <returns> Null if input is not a valid word.
+   /// False if input is a valid wrd but is not the secret word.
+   /// True if input is the secret word.</returns>
+   bool? CompareAndPrint (char[] input) {
+      if (!mDict.Any (x => x.SequenceEqual (string.Join ("", input)))) {
+         Message ("Enter valid word.");
+         return null;
       }
-      #endregion
+      ShiftTo (mStartCol, mRow);
+      for (int i = 0; i < mCursor; i++) {
+         mState[input[i]] = ForegroundColor = input[i] == mSecretWord[i] ? ConsoleColor.Green :

```

Comment:
```suggestion
         if (input[i] == mSecretWord[i]) mState[input[i]] = ForegroundColor = ConsoleColor.Green;
         else if (mSecretWord.Contains (input[i])) {
            if (input[..i].Count (input[i]) >= mSecretWord.Count (input[i])) mState[input[i]] = ForegroundColor = ConsoleColor.DarkGray;
            else mState[input[i]] = ForegroundColor = ConsoleColor.Blue;
         } else mState[input[i]] = ForegroundColor = ConsoleColor.DarkGray;
```
To print only the first 'p' in blue if secret word is 'tempt' and input is 'apple'